src/store/actions.js

```diff
@@ -357,14 +357,40 @@ export default {
 			}
 		}
 
+		// Stop schedule when editing outbox messages and backup sendAt timestamp
+		let originalSendAt
+		if (type === 'outbox' && data.id && data.sendAt) {
+			originalSendAt = data.sendAt
+			const message = {
+				...data,
+				body: data.isHtml ? data.body.value : toPlain(data.body).value,
+			}
+			await dispatch('outbox/stopMessage', { message })
+		}
+
 		commit('showMessageComposer', {
 			type,
 			data,
 			forwardedMessages,
 			templateMessageId,
+			originalSendAt,
 		})
 	},
-	async closeMessageComposer({ commit }) {
+	async closeMessageComposer({ commit, dispatch, getters }, { restoreOriginalSendAt }) {
+		// Restore original sendAt timestamp when requested
+		const message = getters.composerMessage
```

Comment:
The Vuex `composerMessage` holds the *initial value* of the composer when it opens. This operation would override all modifications done before closing the modal.

* Write a new message
* Schedule it
* Reopen
* Make changes
* Close the modal

-> edits are lost

Reply:
https://github.com/nextcloud/mail/issues/6470